### DISCUSSION TITLE
Issue with creating forcing files

### DISCUSSION THREAD

**Discussion by psbecker** [NCAR/hrldas#201] [Q&A]
I'm following the tutorial for the "Compile the create_point_data.f90 code to generate single-point forcing" step in the docker container but when I type the make command, I get:
```
ams@docker-desktop:~/hrldas/hrldas/HRLDAS_forcing/run/examples/single_point$ make
mpif90 -I/usr/include -o create_point_data.exe -g -fconvert=big-endian -fbounds-check -fno-range-check -fallow-argument-mismatch create_point_data.f90 -L/usr/lib -lnetcdf -lnetcdff
```
it still gives me the create_point_data.exe but when I run the executable, my forcing data doesn't have any values in it. I didn't change anything in the user_build_options file because I was using the Docker Container so I'm not sure what's going on

**Comment by cenlinhe** [comment]
When you do "./configure", what compiler option did you choose? You will need to choose this one:
9. Linux gfortran/gcc compiler MPI (Docker container)

**Reply by psbecker** [reply]
I chose option 9.

**Reply by cenlinhe** [reply]
This is weird. Are you running our default single-point forcing data case provided in the github folder? I assume you have the forcing files generated but just without any variables in it? Any error messages?

**Reply by psbecker** [reply]
Correct, using the default sing-point forcing data. I didn't make any changes to it. It'll generate the files but no variables in it. And no error messages

**Reply by psbecker** [reply]
the comile.log gives the same message: I/usr/include -o create_point_data.exe -g -fconvert=big-endian -fbounds-check -fno-range-check -fallow-argument-mismatch create_point_data.f90 -L/usr/lib -lnetcdf -lnetcdff

**Reply by psbecker** [reply]
Hi Cenlin, just wondering if you have been able to look into this more and see what's going on?

**Reply by cenlinhe** [reply]
I just tried to download the NoahMP docker contain to my Macbook and run the single point case for HRLDAS, which works successfully.
![image](https://github.com/user-attachments/assets/e94424c8-aca3-4363-ae65-d5dce922b0de)

I emailed you an example single-point forcing file after execute create_point_data.exe. You did not need to change any setup or hrldas code inside docker container.

**Reply by psbecker** [reply]
I got the same message you did after running the 'make' command with the 'mpif90 -I/usr/include.....' and it was still able to run but the LDASIN_DOMAIN1 files didn't have any values to them. 

For example, when I try to open them with ncview or in jupyter notebooks, it says there is no numeric data. I just tried looking at the example single-point forcing you sent me and I got the same issue. This also happens when I try to open and view one of the output variables. If I use 'ncview 199801020030.LDASOUT_DOMAIN1' I get the issue in the image attached. 
<img width="588" alt="Screenshot 2024-08-18 at 20 41 28" src="https://github.com/user-attachments/assets/7640d07b-dadc-407c-b3f1-553985a06178">

**Reply by cenlinhe** [reply]
This is because it is a point scale input forcing file and output file, and if you use ncview to open it, it will not show the value. Please use the following ncdump command and you will see the numeric value. The file has no issue. I just checked the file I sent to you:
ncdump -v T2D 199801020000.LDASIN_DOMAIN1 
it will show 277.15 K
ncdump -v Q2D 199801020000.LDASIN_DOMAIN1
it will show 0.003747421 kg/kg

etc.

**Reply by cenlinhe** [reply]
note that because it is point scale, every variable is a scalar in hourly input forcing file (similarly for output file, except for the snow and soil layer-related variables). ncview is not a good tool to open scalar variable.

**Comment by psbecker** [comment]
Oh okay. That makes more sense.

I will try this tomorrow and get back to you. Thank you so much for your help! I greatly appreciate it!

Paige

From: Cenlin_He ***@***.***>
Date: Sunday, August 18, 2024 at 20:49
To: NCAR/hrldas ***@***.***>
Cc: Paige Becker ***@***.***>, Author ***@***.***>
Subject: [EXTERNAL] Re: [NCAR/hrldas] Issue with creating forcing files (Discussion #201)
CAUTION: This email originated from outside of the Colorado School of Mines organization. Do not click on links or open attachments unless you recognize the sender and know the content is safe.


note that because it is point scale, every variable is a scalar in hourly input forcing file (similarly for output file, except for the snow and soil layer-related variables). ncview is not a good tool to open scalar variable.

**Comment by psbecker** [comment]
That worked. Thank you Cenlin!

Paige
________________________________
From: Cenlin_He ***@***.***>
Sent: Sunday, August 18, 2024 8:48 PM
To: NCAR/hrldas ***@***.***>
Cc: Paige Becker ***@***.***>; Author ***@***.***>
Subject: [EXTERNAL] Re: [NCAR/hrldas] Issue with creating forcing files (Discussion #201)

CAUTION: This email originated from outside of the Colorado School of Mines organization. Do not click on links or open attachments unless you recognize the sender and know the content is safe.


This is because it is a point scale input forcing file and output file, and if you use ncview to open it, it will not show the value. Please use the following ncdump command and you will see the numeric value. The file has no issue. I just checked the file I sent to you:
ncdump -v T2D 199801020000.LDASIN_DOMAIN1
it will show 277.15 K
ncdump -v Q2D 199801020000.LDASIN_DOMAIN1
it will show 0.003747421 kg/kg

etc.
